### DISCUSSION TITLE
Send field used as query_key to alertmanager

### DISCUSSION THREAD

**Discussion by praneethkaturi** [jertel/elastalert2#977] [Q&A] [answered]
My current alert is configured as follows 

```name: test
type: flatline
index: ops.os4.router*
timeframe:
    minutes: 15
threshold: 1

filter:
- query:
    query_string:
        query: 'NOT (hostname : "default-node" OR hostname : "node-a")'

query_key: "hostname.keyword"
alert_text: "No logs received from {0} in the last 15 minutes"
alert_text_args: ["key"]

alertmanager_labels:
  source: "elastalert"
  severity: "medium"
  ```

I want to add the query_key (which in this case is the hostname) to the alertmanager_labels. This would give me the ability to silence alerts based on the hostname, which is desirable. 

Is there any straightforward way of achieving this?
Thank you in advance!

**Comment by jertel** [comment, marked as answer]
Did you read this portion of the [documentation](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#alertmanager)?

![image](https://user-images.githubusercontent.com/16639611/195616382-84a0e9d0-709e-48ca-860b-83cadb409f23.png)

**Comment by praneethkaturi** [comment]
Hi @jertel sorry for the late reply. Yes I have gone through is and I'm slightly confused about its usage.

alert_text_args I can index the array this way
```
alert_text: "No logs received from {0} in the last 15 minutes"
alert_text_args: ["key"] // I'm assuming this key corresponds to the query_key
```



What would the corresponding way of accessing alertmanager_fields be? Something like this-
```
alertmanager_fields:
    hostname: "@key"
```

**Reply by nsano-rururu** [reply]
@praneethkaturi 
look
https://github.com/jertel/elastalert2/pull/503

**Reply by jertel** [reply]
`alert_text_args: ["hostname"]`?

**Reply by praneethkaturi** [reply]
Hi @nsano-rururu 
Do you want me to look at the following line?

```
alertmanager_fields:
  msg: "message"
  log: "@log_name"
```

Here, is log_name a field in the document?

**Reply by nsano-rururu** [reply]
@praneethkaturi 
log_name is the item that was in the index of elasticsearch when the operation was confirmed. What's wrong with that?

**Reply by praneethkaturi** [reply]
Nothing is wrong with it, but in the case of flatline alerts, the index would not contain the hostname. The hostname is the query_key. For alert_text, I was successfully able to add the hostname as follows:-
```
alert_text: "No logs received from {0} in the last 15 minutes"
alert_text_args: ["key"] // I'm assuming this key corresponds to the query_key
```

I'm confused as to how I would add the label for alertmanager, I tried this:-
```
alertmanager_fields:
  msg: "test"
  log: "@hostname.keyword"
```

But for some reason, both of the fields are null in the alertmanager.

**Reply by nsano-rururu** [reply]
I don't know why different item names are specified for alert_text_args and alertmanager_fields. I think it would be nice to write a key in alertmanager_fields

**Reply by nsano-rururu** [reply]
log: "@hostname.keyword"
 ↓
  log: "hostname.keyword"

**Reply by praneethkaturi** [reply]
My bad, I assumed it had to be represented in a similar way as that of query_key. Simply putting key works.

**Reply by nsano-rururu** [reply]
Problem solved?
